### PR TITLE
Close the session on error in client transports

### PR DESCRIPTION
**Pull request by timostamm**
By default, we close HTTP/2 connections after each request in Node.js. 

But because of an oversight, the code handling the close never ran if we encountered an error, for example an unexpected response content type, or just an error encoded in the response body or response trailers of a streaming RPC.

### CODE REVIEW

File: packages/connect/src/protocol-connect/transport.ts [hunk 1]
```diff
@@ -78,6 +80,7 @@ export function createTransport(opt: CommonTransportOptions): Transport {
         opt.jsonOptions,
         opt
       );
+      const ac = createLinkedAbortController(signal);
```
Comment (timostamm) [inline packages/connect/src/protocol-connect/transport.ts:83]: This is the relevant change: We create an AbortSignal and pass it to the HTTP client, and we wrap the rest of the Transport logic in a try-catch block and abort the signal on error. For HTTP/2, this closes the stream (and session), for HTTP/1.1 it calls destroy() on request and response.

File: packages/connect/src/protocol-connect/validate-response.ts [hunk 2]
```diff
@@ -46,12 +45,7 @@ export function validateResponse(
       `HTTP ${status}`,
       codeFromHttpStatus(status)
     );
-    if (
-      methodKind == MethodKind.Unary &&
-      parsedType &&
-      !parsedType.stream &&
-      !parsedType.binary
-    ) {
+    if (methodKind == MethodKind.Unary && parsedType && !parsedType.stream) {
```
Comment (timostamm) [inline packages/connect/src/protocol-connect/validate-response.ts:48]: This is an unrelated fix. For Connect unary, we always try to parse an error from the response body if we receive a non-successful HTTP status - not just for JSON, but for protobuf binary as well. This has slipped through our test suite so far, but is covered now.

**Review by smaye81 — APPROVED** [review]
